mergify_engine/context.py

```diff
@@ -63,25 +63,74 @@ class Installation:
     subscription: subscription.Subscription
     client: github.GithubInstallationClient
 
+    repositories: "typing.Dict[github_types.GitHubRepositoryName, Repository]" = (
+        dataclasses.field(default_factory=dict)
+    )
+
     @property
     def stream_name(self) -> "worker.StreamNameType":
         return typing.cast(
             "worker.StreamNameType", f"stream~{self.owner_login}~{self.owner_id}"
         )
 
+    def get_repository(self, name: github_types.GitHubRepositoryName) -> "Repository":
+        if name not in self.repositories:
+            repository = Repository(self, name)
+            self.repositories[name] = repository
+        return self.repositories[name]
+
+
+@dataclasses.dataclass
+class Repository(object):
+    installation: Installation
+    name: str
+    pull_contexts: "typing.Dict[github_types.GitHubPullRequestNumber, Context]" = (
+        dataclasses.field(default_factory=dict)
+    )
```

Comment:
Do we need to store this either?

Reply:
This is planned to be replace this Context construction here: https://github.com/Mergifyio/mergify-engine/pull/1729/files#diff-e709bf7f7a0ceb20e0cfafd3b8ab1e163a0c04e84262742607dc6828cf868770R258